flink-staging/flink-table/src/main/scala/org/apache/flink/api/table/codegen/ExpressionCodeGenerator.scala

```diff
@@ -202,6 +210,23 @@ abstract class ExpressionCodeGenerator[R](
           """.stripMargin
         }
 
+      case expressions.Literal(dateValue: Date, DATE_TYPE_INFO) =>
+        val dateName = s"""date_${dateValue.getTime}"""
+        val dateStmt = s"""static java.util.Date $dateName
+             |= new java.util.Date(${dateValue.getTime});""".stripMargin
+        reusableStatements.add(dateStmt)
+
+        if (nullCheck) {
+          s"""
+            |val $nullTerm = false
```

Comment:
Aren't you generating Java code here? This should not work with `val`. I assume that you don't have a test case where you generate this code. Otherwise it should have failed. I guess it would be good to add a test case for it.